From 54facf69e5b1f23d397bd333dd50c200e9c56576 Mon Sep 17 00:00:00 2001
From: T S Rameshkumar <rameshkumar.t@phytecembedded.in>
Date: Fri, 23 Oct 2026 18:17:30 +0530
Subject: [PATCH 14/16] imx8m: phyboard_polis: Enable PF8x00 PMIC regulator
 module

Enable CONFIG_REGULATOR_PF8X00=m in android_addition_defconfig and add
pf8x00-regulator.ko to BOARD_VENDOR_RAMDISK_KERNEL_MODULES.

On PHYTEC phyBOARD-Polis i.MX8MM SOM, the onboard PMIC is NXP PF8121A
controlling LDO2 (reg_nvcc_sd2), which supplies vqmmc for USDHC2 (MicroSD).
Loading pf8x00-regulator.ko after i2c-imx allows 30b50000.mmc to obtain
vqmmc without probe deferral and detect the MicroSD card partitions.

Signed-off-by: T S Rameshkumar <rameshkumar.t@phytecembedded.in>
---
 imx8m/phyboard_polis/SharedBoardConfig.mk       | 7 ++++---
 imx8m/phyboard_polis/android_addition_defconfig | 1 +
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/imx8m/phyboard_polis/SharedBoardConfig.mk b/imx8m/phyboard_polis/SharedBoardConfig.mk
index 498d258c..f1043839 100644
--- a/imx8m/phyboard_polis/SharedBoardConfig.mk
+++ b/imx8m/phyboard_polis/SharedBoardConfig.mk
@@ -123,8 +123,11 @@ BOARD_VENDOR_RAMDISK_KERNEL_MODULES +=     \
     $(KERNEL_OUT)/drivers/clocksource/timer-imx-sysctr.ko \
     $(KERNEL_OUT)/drivers/soc/imx/busfreq-imx8mq.ko \
     $(KERNEL_OUT)/drivers/watchdog/imx2_wdt.ko \
-    $(KERNEL_OUT)/drivers/regulator/pca9450-regulator.ko \
     $(KERNEL_OUT)/drivers/gpio/gpio-mxc.ko \
+    $(KERNEL_OUT)/drivers/i2c/busses/i2c-imx.ko \
+    $(KERNEL_OUT)/drivers/i2c/i2c-dev.ko \
+    $(KERNEL_OUT)/drivers/regulator/pf8x00-regulator.ko \
+    $(KERNEL_OUT)/drivers/regulator/pca9450-regulator.ko \
     $(KERNEL_OUT)/drivers/perf/fsl_imx8_ddr_perf.ko \
     $(KERNEL_OUT)/drivers/mfd/rohm-bd718x7.ko \
     $(KERNEL_OUT)/drivers/regulator/rohm-regulator.ko \
@@ -134,8 +137,6 @@ BOARD_VENDOR_RAMDISK_KERNEL_MODULES +=     \
     $(KERNEL_OUT)/drivers/nvmem/nvmem-imx-ocotp.ko \
     $(KERNEL_OUT)/drivers/mmc/host/sdhci-esdhc-imx.ko \
     $(KERNEL_OUT)/drivers/mmc/host/cqhci.ko \
-    $(KERNEL_OUT)/drivers/i2c/busses/i2c-imx.ko \
-    $(KERNEL_OUT)/drivers/i2c/i2c-dev.ko \
     $(KERNEL_OUT)/drivers/spi/spidev.ko \
     $(KERNEL_OUT)/drivers/spi/spi-bitbang.ko \
     $(KERNEL_OUT)/drivers/spi/spi-nxp-fspi.ko \
diff --git a/imx8m/phyboard_polis/android_addition_defconfig b/imx8m/phyboard_polis/android_addition_defconfig
index 039d0435..6133f796 100644
--- a/imx8m/phyboard_polis/android_addition_defconfig
+++ b/imx8m/phyboard_polis/android_addition_defconfig
@@ -1,4 +1,5 @@
 # PHYTEC phyBOARD-Polis i.MX8MM additions
+CONFIG_REGULATOR_PF8X00=m
 CONFIG_DP83867_PHY=m
 CONFIG_RTC_DRV_RV3028=m
 CONFIG_LOCALVERSION="-g758ad1ac4e8e-PHYTEC-Android15"
-- 
2.34.1

